mne/io/base.py

```diff
@@ -1450,6 +1451,72 @@ def save(self, fname, picks=None, tmin=0, tmax=None, buffer_size_sec=None,
                    start, stop, buffer_size, projector, drop_small_buffer,
                    split_size, split_naming, 0, None, overwrite)
 
+    def export(self, fname, fmt="auto"):
+        """Export Raw to external formats.
+
+        Supported formats: EEGLAB (set)
+
+        .. warning::
+            Since we are exporting to external formats, there's no
+            guarantee that all the info will be preserved in the external
+            format. To save in native MNE format (fif) without info loss,
+            use :func:`save` instead.
+
+        Parameters
+        ----------
+        fname : str
+            Name of the export file.
+        fmt : str
+            Format of the export. Defaults to auto, which will infer
+            format from fname's extension.
+
+        Notes
+        -----
+        For EEGLAB format, channel locations are expanded to full EEGLAB
+        format. For more details see :func:`eeglabio.utils.cart_to_eeglab`.
+        """
+        if fmt == "auto":
+            fmt = op.splitext(fname)[1]
+            if fmt:
+                fmt = fmt[1:]
+            else:
+                raise ValueError("Couldn't infer format from filename "
+                                 "(no extension found)")
+
+        fmt = fmt.lower()
+
+        if fmt == "set":
+            _check_eeglabio_installed()
+            from eeglabio.raw import export_set
+            # load data first
+            self.load_data()
+
+            # remove extra epoc and STI channels
+            drop_chs = ['epoc']
+            if not (self.filenames[0].endswith('.fif')):
+                drop_chs.append('STI 014')
+            pick_chs = [ch for ch in self.ch_names if ch not in drop_chs]
+
+            chs = self.info["chs"]
+            cart_coords = np.array([d['loc'][:3] for d in chs
+                                    if d['ch_name'] not in drop_chs])
+            if cart_coords.any():  # has coordinates
+                # (-y x z) to (x y z)
+                cart_coords[:, 0] = -cart_coords[:, 0]  # -y to y
+                # swap x (1) and y (0)
+                cart_coords[:, [0, 1]] = cart_coords[:, [1, 0]]
+            else:
+                cart_coords = None
+
+            annotations = [self.annotations.description,
+                           self.annotations.onset,
+                           self.annotations.duration]
+            export_set(fname, self.get_data(picks=pick_chs),
+                       self.info['sfreq'],
+                       pick_chs, cart_coords, annotations)
+        else:
+            raise ValueError("Format not supported (%s)" % fmt)
```

Comment:
too much copy paste @jackz314 I would have to type my comments here too.

can you avoid this with a util private function?

Reply:
moved to io.uitls and utils.check